ui/packages/catalog-ui-search/src/main/webapp/react-component/location/bounding-box.js

```diff
@@ -74,34 +93,129 @@ const BoundingBoxLatLon = props => {
       />
       <TextField
         label="North"
-        value={north}
-        onChange={cursor('north')}
+        value={north !== undefined ? String(north) : north}
+        onChange={value => onChangeDd('north', value)}
+        onBlur={() => setDdError(validateGeo('north', north))}
         type="number"
         step="any"
         min={northMin || -90}
         max={90}
         addon="°"
       />
+      {getErrorComponent(ddError)}
+    </div>
+  )
+}
+
+const BoundingBoxLatLonDms = props => {
+  const {
+    dmsSouth,
+    dmsNorth,
+    dmsWest,
+    dmsEast,
+    dmsSouthDirection,
+    dmsNorthDirection,
+    dmsWestDirection,
+    dmsEastDirection,
+    setState,
+  } = props
+  const [dmsError, setDmsError] = useState({
+    error: false,
+    errorMsg: '',
+    defaultValue: '',
+  })
+  const latitudeDirections = [Direction.North, Direction.South]
+  const longitudeDirections = [Direction.East, Direction.West]
+
+  function validate(key, type, value) {
+    const { error, message, defaultValue } = validateGeo(key, value)
+    if (type === 'blur') {
+      setDmsError({
+        error: value !== undefined && value.length === 0,
+        message,
+        defaultValue,
+      })
+    } else if (defaultValue) {
+      setDmsError({
+        error,
+        message,
+        defaultValue,
+      })
+    }
+    defaultValue ? setState(key, defaultValue) : setState(key, value)
```

Comment:
Could put `setState(key, value)` in the `if` and `setState(key, defaultValue)` in the `else if`

Reply:
scratch that